cmd/account/account-assign.go

```
@@ -0,0 +1,147 @@
+	if o.username == "" {
+		return cmdutil.UsageErrorf(cmd, "RedHat username was not provided")
+	}
+
+	var err error
+	o.kubeCli, err = k8s.NewClient(o.flags)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *accountAssignOptions) run() error {
+
+	var (
+		accountAssignID string
+	)
+
+	if o.username != "" && o.payerAccount == "" {
+		o.payerAccount = "osd-staging-2"
+	}
+	//Instantiating aws client
+	AwsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+	//Listing accounts
+	input := &organizations.ListAccountsInput{}
+	accounts, err := AwsClient.ListAccounts(input)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	//Lopping through the list to find an unassigned account
+	for _, a := range accounts.Accounts {
+		if *a.Status == "UNASSIGNED" {
+			accountAssignID = *a.Id
+			break
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	//Tagging the account with username
+	inputTag := &organizations.TagResourceInput{
+		ResourceId: aws.String(accountAssignID),
+		Tags: []*organizations.Tag{
+			{
+				Key:   aws.String(accountAssignID),
+				Value: aws.String(o.username),
+			},
+		},
+	}
+
+	_, err = AwsClient.TagResource(inputTag)
+	if err != nil {
+		return err
+	}
+
+	//Moving to developers OU
+	inputMove1 := &organizations.MoveAccountInput{
+
+		AccountId:           aws.String(accountAssignID),
+		DestinationParentId: aws.String("ou-0wd6-z6tzkjek"),
+		SourceParentId:      aws.String("r-0wd6"),
+	}
+	inputMove2 := &organizations.MoveAccountInput{
+		AccountId:           aws.String(accountAssignID),
+		DestinationParentId: aws.String("ou-rs3h-i0v69q47"),
+		SourceParentId:      aws.String("r-0wd6"),
+	}
+
+	if o.payerAccount == "osd-staging-2" {
+		_, err = AwsClient.MoveAccount(inputMove2)
+		if err != nil {
+			return err
+		}
+	} else if o.payerAccount == "osd-staging-1" {
+		_, err = AwsClient.MoveAccount(inputMove1)
+		if err != nil {
+			fmt.Println(err.Error())
```

should we continue if there is an error? if not, then we should return the err here.